### PR TITLE
fix: truncate log item strings

### PR DESCRIPTION
**Pull request by jonaro00**

### CODE REVIEW

**Review by chesedo — APPROVED** [review]
LGTM

**Review by orhun — APPROVED** [review]

File: common/src/log.rs [hunk 1]
```diff
@@ -54,8 +54,15 @@ pub struct LogItem {
     pub line: String,
 }
 
+const LOG_LINE_MAX_CHARS: usize = 2048;
+
 impl LogItem {
-    pub fn new(id: Uuid, internal_origin: Backend, line: String) -> Self {
+    pub fn new(id: Uuid, internal_origin: Backend, line: impl Into<String>) -> Self {
+        let mut line = line.into();
+        if line.chars().count() > LOG_LINE_MAX_CHARS {
```
Comment (oddgrd) [inline common/src/log.rs:62]: Unless I am reading this incorrectly we now iterate over all the chars each time we create a new LogItem. Should we rather check the `len()`, which is stored with the string?

File: common/src/log.rs [hunk 2]
```diff
@@ -54,8 +54,15 @@ pub struct LogItem {
     pub line: String,
 }
 
+const LOG_LINE_MAX_CHARS: usize = 2048;
+
 impl LogItem {
-    pub fn new(id: Uuid, internal_origin: Backend, line: String) -> Self {
+    pub fn new(id: Uuid, internal_origin: Backend, line: impl Into<String>) -> Self {
```
Comment (oddgrd) [inline common/src/log.rs:60]: Can't this be just `line: String`?

File: common/src/log.rs (hunk 1 above)
Comment (oddgrd) [reply common/src/log.rs:62]: And if the line is over the limit, we still need to iterate over the entire string before we can know the count is over the limit.

File: common/src/log.rs [hunk 3]
```diff
@@ -54,8 +54,15 @@ pub struct LogItem {
...
+        let mut line = line.into();
+        if line.chars().count() > LOG_LINE_MAX_CHARS {
+            line = line.chars().take(LOG_LINE_MAX_CHARS).collect()
```
Comment (oddgrd) [inline common/src/log.rs:63]: Should we also add something to make it clear that this is truncated, like `...`?

Comment (iulianbarbu) [reply common/src/log.rs:63]: I would add a suffix with `... [ truncated ]`.

File: common/src/log.rs (hunk 2 above)
Comment (jonaro00) [reply common/src/log.rs:60]: impl into is generally friendlier in an API that consumes String.

**Pull request merged**